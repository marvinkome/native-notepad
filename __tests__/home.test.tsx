/**
 * __tests__/home.test
 */

import * as React from 'react';
import renderer from 'react-test-renderer';
import Home from '../src/components/home';

it('Renders homepage correctly', () => {
    const hello = renderer.create(<Home />).toJSON();
    expect(hello).toMatchSnapshot();
});
