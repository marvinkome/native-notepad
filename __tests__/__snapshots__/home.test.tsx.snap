// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders dumb homepage correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
    virtual={undefined}
  >
    <View>
      <View
        closeOnRowBeginSwipe={false}
        closeOnRowPress={true}
        closeOnScroll={true}
        dataArray={undefined}
        directionalDistanceChangeThreshold={2}
        disableLeftSwipe={false}
        disableRightSwipe={false}
        leftOpenValue={0}
        previewFirstRow={false}
        recalculateHiddenLayout={false}
        renderRow={[Function]}
        rightOpenValue={0}
        style={Object {}}
        swipeToOpenPercent={50}
        virtual={undefined}
      />
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "alignItems": "center",
          "bottom": 20,
          "flexDirection": "column",
          "height": 56,
          "left": undefined,
          "position": "absolute",
          "right": 20,
          "top": undefined,
          "width": 56,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "blue",
            "borderRadius": 28,
            "bottom": 0,
            "elevation": 4,
            "height": 56,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 2,
            "width": 56,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#fff",
                  "fontSize": 24,
                  "lineHeight": 27,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          virtual={undefined}
        >
          ï„‚
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Renders homepage correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
    virtual={undefined}
  >
    <View>
      <RCTScrollView
        closeOnRowBeginSwipe={false}
        closeOnRowPress={true}
        closeOnScroll={true}
        dataArray={
          Array [
            Object {
              "body": "This is a notepad app",
              "category": 1,
              "date": 0,
              "id": "0",
              "title": "Note 1",
            },
            Object {
              "body": "This is the second note",
              "category": 1,
              "date": 0,
              "id": "1",
              "title": "Note 2",
            },
          ]
        }
        dataSource={
          ListViewDataSource {
            "items": 2,
          }
        }
        directionalDistanceChangeThreshold={2}
        disableLeftSwipe={false}
        disableRightSwipe={false}
        enableEmptySections={true}
        leftOpenValue={0}
        previewFirstRow={false}
        recalculateHiddenLayout={false}
        renderRow={[Function]}
        renderScrollComponent={[Function]}
        rightOpenValue={0}
        style={Object {}}
        swipeToOpenPercent={50}
        virtual={undefined}
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderBottomColor": "transparent",
                    "borderBottomWidth": 0.5,
                    "flexDirection": "row",
                    "marginLeft": 0,
                    "paddingBottom": 15,
                    "paddingRight": 18,
                    "paddingTop": 15,
                  },
                  Object {
                    "borderLeftColor": "red",
                    "borderLeftWidth": 7,
                    "marginBottom": 1,
                  },
                ]
              }
              virtual={undefined}
            >
              <View
                style={
                  Object {
                    "alignItems": null,
                    "alignSelf": null,
                    "flex": 1,
                  }
                }
                virtual={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "Roboto",
                        "fontSize": 16,
                        "marginLeft": 12,
                        "marginRight": 12,
                      },
                      Object {
                        "fontSize": 23,
                        "fontWeight": "600",
                      },
                    ]
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  Note 1
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "alignSelf": "center",
                    "flex": 0.28,
                    "paddingBottom": null,
                    "paddingLeft": null,
                    "paddingRight": null,
                    "paddingTop": null,
                  }
                }
                virtual={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": null,
                        "color": "#000",
                        "fontFamily": "Roboto",
                        "fontSize": 16,
                      },
                      Object {
                        "fontSize": 13,
                        "fontWeight": "200",
                      },
                    ]
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  11/12/18
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderBottomColor": "transparent",
                    "borderBottomWidth": 0.5,
                    "flexDirection": "row",
                    "marginLeft": 0,
                    "paddingBottom": 15,
                    "paddingRight": 18,
                    "paddingTop": 15,
                  },
                  Object {
                    "borderLeftColor": "red",
                    "borderLeftWidth": 7,
                    "marginBottom": 1,
                  },
                ]
              }
              virtual={undefined}
            >
              <View
                style={
                  Object {
                    "alignItems": null,
                    "alignSelf": null,
                    "flex": 1,
                  }
                }
                virtual={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "Roboto",
                        "fontSize": 16,
                        "marginLeft": 12,
                        "marginRight": 12,
                      },
                      Object {
                        "fontSize": 23,
                        "fontWeight": "600",
                      },
                    ]
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  Note 2
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "alignSelf": "center",
                    "flex": 0.28,
                    "paddingBottom": null,
                    "paddingLeft": null,
                    "paddingRight": null,
                    "paddingTop": null,
                  }
                }
                virtual={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": null,
                        "color": "#000",
                        "fontFamily": "Roboto",
                        "fontSize": 16,
                      },
                      Object {
                        "fontSize": 13,
                        "fontWeight": "200",
                      },
                    ]
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  11/12/18
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "alignItems": "center",
          "bottom": 20,
          "flexDirection": "column",
          "height": 56,
          "left": undefined,
          "position": "absolute",
          "right": 20,
          "top": undefined,
          "width": 56,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "blue",
            "borderRadius": 28,
            "bottom": 0,
            "elevation": 4,
            "height": 56,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 2,
            "width": 56,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#fff",
                  "fontSize": 24,
                  "lineHeight": 27,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          virtual={undefined}
        >
          ï„‚
        </Text>
      </View>
    </View>
  </View>
</View>
`;
